### PR TITLE
Initial refactoring and addition of plugins

### PR DESCRIPTION
**Pull request by duncannixon**
Added 'use strict' to the top of all .js files in order to prevent the accidental use of variables that haven't been explicitly declared.

Standardised the treatment of Hapi.js Plugins. Added the following new plugins: disinfect, hapi-robots, hapi-sanitize-payload

Renamed JS files, adding .plugin, .route or .service to filenames in order to give more information on what type of JS files they are.

Refactored - added Views collection in constants.Views to avoid duplication of strings when defining the views for routes.

Added IDs to continue buttons to facilitate unit and automated testing.

Made the back links work (via the govukBackLink component)

### CODE REVIEW

File: server/routes/ivory-added.route.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        }
+      })
+    } else if (payload.yesNoIdk === 'No') {
+      const client = request.redis.client
+      client.set('ivory-added', 'no')
+      return h.redirect('check-your-answers')
+    } else if (payload.yesNoIdk === 'I dont know') {
+      return 'Game over man...game over!'
+    } else if (payload.yesNoIdk === 'Yes') {
+      return h.redirect('taken-from-elephant')
```
Comment (dgretho) [inline server/routes/ivory-added.route.js:37]: This should be Views.TAKEN_FROM_ELEPHANT

File: server/routes/ivory-integral.route.js [hunk 2]
```diff
@@ -1,19 +1,25 @@
...
       errorText: false
     })
   },
+
   post: (request, h) => {
     const payload = request.payload
     if (payload.ivoryIsIntegral) {
       const client = request.redis.client
       client.set('ivory-integral', payload.ivoryIsIntegral)
       return h.redirect('check-your-answers')
```
Comment (dgretho) [inline server/routes/ivory-integral.route.js:18]: This wants to be Views.CHECK_YOUR_ANSWERS

File: server/routes/taken-from-elephant.route.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+module.exports = [
+  {
+    method: 'GET',
+    path: '/taken-from-elephant',
+    handler: handlers.get
+  },
+  {
+    method: 'POST',
+    path: '/taken-from-elephant',
```
Comment (dgretho) [inline server/routes/taken-from-elephant.route.js:49]: Views.TAKEN_FROM_ELEPHAT

File: server/routes/taken-from-elephant.route.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    } else if (payload.yesNoIdk === 'Yes') {
+      return 'Those poor elephants, you are not selling that!'
+    }
+  }
+}
+
+module.exports = [
+  {
+    method: 'GET',
+    path: '/taken-from-elephant',
```
Comment (dgretho) [inline server/routes/taken-from-elephant.route.js:44]: Views.TAKEN_FROM_ELEPHANT

File: server/routes/taken-from-elephant.route.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+          'You must tell us if the replacement ivory was taken from an elephant on or after 1 January 1975',
+        errorText: {
+          text:
+            'You must tell us if the replacement ivory was taken from an elephant on or after 1 January 1975'
+        }
+      })
+    } else if (payload.yesNoIdk === 'No') {
+      const client = request.redis.client
+      client.set('ivory-added', 'yes-pre-1975')
+      return h.redirect('check-your-answers')
```
Comment (dgretho) [inline server/routes/taken-from-elephant.route.js:32]: Views.CHECK_YOUR_ANSWERS

File: server/routes/taken-from-elephant.route.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+      errorSummaryText: '',
+      errorText: false
+    })
+  },
+
+  post: (request, h) => {
+    const payload = request.payload
+
+    if (!payload.yesNoIdk) {
+      return h.view('yes-no-idk', {
```
Comment (dgretho) [inline server/routes/taken-from-elephant.route.js:18]: Views.YES_NO_IDK

File: server/routes/taken-from-elephant.route.js [hunk 7]
```diff
@@ -0,0 +1,52 @@
+'use strict'
+
+const handlers = {
+  get: (request, h) => {
+    return h.view('yes-no-idk', {
```
Comment (dgretho) [inline server/routes/taken-from-elephant.route.js:5]: Views.YES_NO_IDK

**Review by dgretho — APPROVED** [review]
Looks good. Just commented on a few more things that could have been moved into Views constants

File: server/routes/taken-from-elephant.route.js (hunk 6 above)
Comment (duncannixon) [reply server/routes/taken-from-elephant.route.js]: Fixed.

File: server/routes/taken-from-elephant.route.js (hunk 5 above)
Comment (duncannixon) [reply server/routes/taken-from-elephant.route.js]: Fixed.

File: server/routes/taken-from-elephant.route.js (hunk 4 above)
Comment (duncannixon) [reply server/routes/taken-from-elephant.route.js]: Fixed.

File: server/routes/taken-from-elephant.route.js (hunk 3 above)
Comment (duncannixon) [reply server/routes/taken-from-elephant.route.js]: Fixed.

File: server/routes/ivory-integral.route.js (hunk 2 above)
Comment (duncannixon) [reply server/routes/ivory-integral.route.js]: Fixed.

File: server/routes/ivory-added.route.js (hunk 1 above)
Comment (duncannixon) [reply server/routes/ivory-added.route.js]: Fixed.

File: server/routes/taken-from-elephant.route.js (hunk 7 above)
Comment (duncannixon) [reply server/routes/taken-from-elephant.route.js]: Fixed.

**Review by Jozzey — COMMENTED** [review]
Looks like we are using the view constants for some of the paths. This might work alright in most cases where the view and path have the same name but thats not always the case where views are re-used. If we are going to use constants for the paths shouldn't these be separate from the ones we are using for the views?

**Comment by duncannixon** [comment]
I have added a new Paths constant to allow views and paths to differ or be reused.

File: server/utils/constants.js [hunk 8]
```diff
@@ -0,0 +1,19 @@
+const Constants = (module.exports = {})
+
+Constants.Paths = {
+  CHECK_YOUR_ANSWERS: 'check-your-answers',
+  HOME: 'home',
+  IVORY_ADDED: 'ivory-added',
+  IVORY_INTEGRAL: 'ivory-integral',
+  TAKEN_FROM_ELEMENT: 'taken-from-elephant',
+  YES_NO_IDK: 'yes-no-idk'
```
Comment (Jozzey) [inline server/utils/constants.js:9]: Just a minor thing. We don't need a path YES_NO_IDK & the path for HOME is just / so can probably get away with just removing them both?

Comment (duncannixon) [reply server/utils/constants.js:9]: Well spotted, removed.

**Pull request merged**